Using Custom Policies article (#50)
ShiftLeftSecurity/documentation: using-inspect-protect/using-cli/cli-reference.md

sl [global options] command [command options] [arguments...]
```

## `sl` Commands

Command | Description
--- | ---
`auth` | [Authenticate the CLI with your ShiftLeft account](../using-cli/authenticating.md).
`analyze [<path>]` | [Use ShiftLeft Inspect to analyze your application](../../using-inspect-protect/analyzing-applications-in-ci.md).  `<path>` can be the path to a `.jar`, `.war` or `.ear` file, or it can be the path to a Java project directory. If `<path>` is not provided, then `.` is implied.
`help`, `h` | List ShiftLeft CLI commands or help for one command.
`install [dotnet-agent]` | Runs the ShiftLeft Protect for .NET Microagent installer.
`policy assignment list` | Lists all security policy assignments for your organization or project that you have previously identified for use with ShiftLeft Inspect.

> `policy assignment remove [--project myprojectname]` | Removes the default security policy of your organization or (if the `project` parameter is provided) your organization's project. 

Reviewer: ```suggestion
`policy assignment remove [--project <project-name>]` | Removes the default security policy of your organization or (if the `project` parameter is provided) your organization's project. 
```

> `policy assignment set mypolicy[:tag] [--project myprojectname]` | Specifies the default security policies used by ShiftLeft Inspect. `set` allows you to assign a default security policy used by ShiftLeft Inspect moving forward. If the optional `project` parameter is provided, the default security policy is only applied when ShiftLeft Inspect analyses the application specified by the `myprojectname` argument. If the `project` parameter is missing, the security policy is applied globally to all applications in your organization.

Reviewer: ```suggestion
`policy assignment set <policy-label>[:<policy-tag>] [--project <project-name>]` | Specifies the default security policies used by ShiftLeft Inspect. `set` allows you to assign a default security policy used by ShiftLeft Inspect moving forward. If the optional `project` parameter is provided, the default security policy is only applied when ShiftLeft Inspect analyses the application specified by the `project-name` argument. If the `project` parameter is missing, the security policy is applied globally to all applications in your organization.
```

`policy create [default\|\no-dictionary] [path-to-policy-file]` | Creates a custom security policy from a pre-defined template. Currently two templates are available: default and no-dictionary. The `default` template creates a policy that imports all standard definitions as used by ShiftLeft, including the generic dictionary of sensitive-data variables. The `no-dictionary` template excludes these standard definitions. A new custom security policy is stored in the file defined by the `path-to-policy-file` argument, or printed to standard output if the argument is omitted.

> `policy info [[domain/]mypolicy[:tag]]` | Returns meta-information about all security policies with the name specified by `mypolicy` argument. If the policy name is omitted, meta-information of all policies available to your organization is returned. The `domain` and `tag` elements are optional. `domain` refers to the policy domain available to your organization. By default it is your organization’s ID, but it could also be publicly available security policies such as `io.shiftleft`. If the `tag` element is omitted, by default the response includes all authorized security policies.

Reviewer: ```suggestion
`policy info [[<policy-domain>/]<policy-label>[:<policy-tag>]]` | Returns meta-information about all security policies with the name specified by `<policy-label>` argument. If the policy name is omitted, meta-information of all policies available to your organization is returned. The `policy-domain` and `policy-tag` elements are optional. `policy-domain` refers to the policy domain available to your organization. By default it is your organization’s ID, but it could also be publicly available security policies such as `io.shiftleft`. If the `policy-tag` element is omitted, by default the response includes all authorized security policies.
```

`policy pull --hash <policy-hash> [path-to-policy-file]` | Fetches the security policy, with the hash specified by  the `policy-hash` argument, from the ShiftLeft infrastructure, if authorized. If the `path-to-policy-file` argument is missing, the security policy is printed to the console. Returns the policy as a text file or a bad request with detailed error information.

> `policy pull mypolicy[:tag] [path-to-policy-file]` | Fetches the security policy, with the name specified by the `mypolicy` argument, from the ShiftLeft infrastructure, if authorized. Use of the policy tag is optional (for example `0.0.1` in `mypolicy:0.0.1`); by default the value is the latest policy. If the `path-to-policy-file` argument is missing, the security policy is printed to the console. Returns the policy as a text file or a bad request with detailed error information.

Reviewer: ```suggestion
`policy pull <policy-label>[:tag] [path-to-policy-file]` | Fetches the security policy, with the name specified by the `policy-label` argument, from the ShiftLeft infrastructure, if authorized. Use of the policy tag is optional (for example `0.0.1` in `mypolicy:0.0.1`); by default the value is the latest policy. If the `path-to-policy-file` argument is missing, the security policy is printed to the console. Returns the policy as a text file or a bad request with detailed error information.
```

> `policy push mypolicy[:tag] ~/path/to/policy.policy` | Uploads a custom security policy, with the name specified by the `mypolicy` argument, to the ShiftLeft policy repository. Use of the policy tag is optional (for example `0.0.1` in `mypolicy:0.0.1`); by default the value is the latest policy. Returns either OK or a bad request with detailed error information.

Reviewer: ```suggestion
`policy push <policy-label>[:<policy-tag>] ~/path/to/policy.policy` | Uploads a custom security policy, with the name specified by the `policy-label` argument, to the ShiftLeft policy repository. Use of the policy tag is optional (for example `0.0.1` in `mypolicy:0.0.1`); by default the value is the latest policy. Returns either OK or a bad request with detailed error information.
```

> `policy verify ~/path/to/policy.policy` | Verifies a custom security policy by sending it to the ShiftLeft infrastructure, which checks for syntactic and semantic errors. Returns either OK or a bad request with detailed error information.

Reviewer: ```suggestion
`policy verify <path-to-policy-file>` | Verifies a custom security policy by sending it to the ShiftLeft infrastructure, which checks for syntactic and semantic errors. Returns either OK or a bad request with detailed error information.
```

> `push <path> [<path>...]` | (Currently not implemented) Upload policies to ShiftLeft.

Reviewer: That should be removed, since we now describe those commands

Author: Done
